lib/codegen/src/verifier/mod.rs

```diff
@@ -120,15 +174,58 @@ impl Display for VerifierError {
 }
 
 /// Verifier result.
-pub type VerifierResult<T> = Result<T, VerifierError>;
+pub type VerifierResult<T> = Result<T, ()>;
```

Comment:
Thought experiment: What if we split `VerifierResult` into two, like
```rust
type VerifierResult<T> = Result<T, VerifierErrors>;
type VerifierPartialResult<T> = Result<T, ()>;
```
and then made every function that currently uses `VerifierResult` *either*:
 - use `VerifierPartialResult` and take an &mut `VerifierResults`, or
 - use `VerifierResult` and don't take an &mut `VerifierResults`
?